discovery/consul/consul.go

```diff
@@ -414,103 +420,100 @@ func (d *Discovery) watchService(ctx context.Context, ch chan<- []*targetgroup.G
 	}
 
 	go func() {
+		ticker := time.NewTicker(d.refreshInterval)
 		var lastIndex uint64
-		for range ticker.C {
-			srv.watch(ctx, ch, &lastIndex)
+		catalog := srv.client.Catalog()
+		for ; true; <-ticker.C {
```

Comment:
You should be checking the context here

Reply:
isn't it sufficient to check ticker.C ? (which is going to be closed when the context is done)